client/src/main/java/org/eclipse/hono/client/AuthenticationServerClient.java

```diff
@@ -90,7 +92,9 @@ public void verifyPlain(final String authzid, final String authcid, final String
         options.addEnabledSaslMechanism(AuthenticationConstants.MECHANISM_PLAIN);
         factory.connect(options, authcid, password, null, null, conAttempt -> {
             if (conAttempt.failed()) {
-                authenticationResultHandler.handle(Future.failedFuture("cannot connect to Authentication service"));
+                authenticationResultHandler
+                        .handle(Future.failedFuture(new ServerErrorException(HttpURLConnection.HTTP_UNAVAILABLE,
```

Comment:
Is there a reason why we *swallow* the root cause for the failure to connect? The attempt might have been failed with a `javax.security.sasl.AuthenticationException` or a `io.vertx.proton.sasl.SaslSystemException` so we might be able to fail the future with a much more specific exception ...

Reply:
Right, now we can differentiate further. The mechanism to get the correct `ServiceInvocationException` would be the same as in `HonoClientImpl#failConnectionAttempt`. 
I've adopted it from there, with modified exception messages.
(It might make sense to use a common method, I'm not sure where to put that though. And we might want to keep the different exception messages.)